### PR TITLE
proposal: submariner as fleet plugin

### LINKED ISSUE #657: Add submariner as fleet plugin
**What would you like to be added**:
Expand the fleet manager's capabilities to install submariner as a Fleet plugin.

**Why is this needed**:
To enhance the user experience and the robustness of the installation process

### PR DESCRIPTION
**Pull request by Flying-Tom**
**What type of PR is this?**

/kind api-change
/kind feature

**What this PR does / why we need it**:

Expand the fleet manager's capabilities to install submariner as a Fleet plugin.

**Which issue(s) this PR fixes**:
Fixes #657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/proposals/submariner/submariner_as_fleet_plugin.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ //   version: 0.14.9
+ //   targetNamespace: submariner
+ // ```
+ //
+ // +optional
+ Chart *ChartConfig `json:"chart,omitempty"`
+ // ExtraArgs is the set of extra arguments for submariner, and example will be provided in the future.
+ //
+ // +optional
+ ExtraArgs apiextensionsv1.JSON `json:"extraArgs,omitempty"`
```
Comment (LiZhenCheng9527) [inline docs/proposals/submariner/submariner_as_fleet_plugin.md:64]: If you wish to leave some custom fields, then you will also need to add a value field

**Review by LiZhenCheng9527 — COMMENTED** [review]
Can you introduce about submariner? And purpose of adding submarine to kurator and what you want to do with it

**Review by zirain — APPROVED** [review]
/lgtm

**Pull request merged**